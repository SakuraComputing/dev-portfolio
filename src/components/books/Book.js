import React from 'react';

const Book = (props) => {
    return (
        <div>
            <div className="item">
                <h4 className="title"><a href="#">{props.book.title}</a> <span
                    className="label label-theme">{props.book.author}</span></h4>
                <p className="summary">{props.book.description}</p>
                <p className="summary">{props.book.author}</p>
                <p><a className="more-link"
                      href={props.book.link}
                      target="_blank"><i className="fa fa-external-link"></i> Purchase here</a></p>
            </div>

        </div>
    );
};

export default Book;
