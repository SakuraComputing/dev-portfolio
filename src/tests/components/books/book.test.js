import React from 'react';
import { shallow } from 'enzyme';
import Book from '../../../components/books/Book';

describe('Book Section', () => {
    it('should render correctly', () => {
        const book = [
            {
                "id": 0,
                "title": "Survive JS - Webpack and React",
                "description": "A guide to programming in React and Webpack",
                "author": "Juho Vepsäläinen",
                "link": "http://leanpub.com/survivejs_webpack_react"
            }
        ];
        const wrapper = shallow(<Book book={book}/>);
        expect(wrapper).toMatchSnapshot();
    });
});
