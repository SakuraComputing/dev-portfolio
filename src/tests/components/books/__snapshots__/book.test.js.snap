// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Book Section should render correctly 1`] = `
<div>
  <div
    className="item"
  >
    <h3
      className="title"
    >
      <a
        href="#"
      />
       
      <span
        className="label label-theme"
      />
    </h3>
    <p
      className="summary"
    />
    <p
      className="summary"
    />
    <p>
      <a
        className="more-link"
        target="_blank"
      >
        <i
          className="fa fa-external-link"
        />
         Purchase here
      </a>
    </p>
  </div>
</div>
`;
